src/Compilers/CSharp/Portable/CSharpParseOptions.cs

```diff
@@ -184,11 +135,6 @@ public new CSharpParseOptions WithDocumentationMode(DocumentationMode documentat
                 return this;
             }
 
-            if (!documentationMode.IsValid())
-            {
-                throw new ArgumentOutOfRangeException(nameof(documentationMode));
-            }
-
             return new CSharpParseOptions(this) { DocumentationMode = documentationMode };
```

Comment:
I think we are moving towards the same model we have with CompilationOptions. I.e. construction doesn't throw. Never. Instead errors are surfaced as diagnostics. For example, Platform an enum, OutputKind is an enum, OptimizationLevel is an enum, but compilation options does not throw on invalid values for them.